content/learn/book/ecs/_index.md

```diff
@@ -0,0 +1,140 @@
++++
+title = "Entities, components and systems"
+weight = 2
+sort_by = "weight"
+template = "book-section.html"
+page_template = "book-section.html"
++++
+
+Bevy is fundamentally powered by its ECS (Entity Component System): almost all data is stored as components on entities, and all logic is executed by its systems.
+
+As we mentioned in the last chapter, all of our data is stored in a [`World`](https://docs.rs/bevy/latest/bevy/ecs/world/struct.World.html) on our [`App`](https://docs.rs/bevy/latest/bevy/app/struct.App.html).
+We can think of our **entity-component data storage** as a giant database:
+
+* each row is an **entity**, representing an object (perhaps a player, tile, or UI button) in our game
+* each column is a type of **component**, storing data of a particular type (perhaps the sprite, team or life of a player entity) in an efficient way
+* each cell is a component of a particular entity, which has a concrete value we can look up and change
+* we access data from this database using **queries**, which fetch entities with the specified components
+* the primary key of this database is the [`Entity`](https://docs.rs/bevy/latest/bevy/ecs/entity/struct.Entity.html) identifier, which can use to look up specific entities using [`Query::get(my_entity)`](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.Query.html#method.get)
+
+Of course, this database is not very well-normalized: not all entities will have every component!
+We can use this to specialize behavior between entities: only performing work on them in our systems if they have the correct combination of components.
+You don't want to apply gravity to entities without a position in your world, and you're only interested in using the UI layout algorithm to control the layout of UI entities!
+
+When we want to go beyond this tabular data storage, we can use **resources**: global singletons which store data in monolithic blobs.
+You might use resources to store one-off bits of state like the game's score, use it to interface with other libraries, or store secondary data structures like indexes to augment your use of entity-component data.
+Just like with components on entities, resources are accessed by type and you can only have one resource of each Rust type.
+
+In order to actually perform logic on all of this data, we must use systems, which are automatically run by our scheduler.
+**Systems** are Rust functions which request specific data from the world, as declared in their system parameters (function arguments): generally resources and entities that have a particular combination of components using queries.
```

Comment:
- I agree that "using queries" is weird here. However, I think that whole elaboration of *what* data systems might request could be left out, since..

- I think this explanation doesn't really give the reader an idea of what the *purpose* of a system is. 

  As a reader, why do I care about Systems? What does this feature let me do?
  
  My suggestion would be to mention the purpose a bit more: perhaps something like:
  
  > **Systems** are Rust functions that request data from the World, in order to perform tasks - often a system is a procedure that specifies how entities may interact. They prescribe all of the rules and behaviours of the World.

Reply:
Adapted your suggestion <3 I agree that there needed to be more *intent* present.

> Systems are Rust functions that request data from the `World` (such as resources or entities with particular components) in order to perform tasks.
> All of the rules and behaviours of our game are governed by systems.